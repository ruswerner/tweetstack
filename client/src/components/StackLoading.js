import React, {PureComponent} from 'react';
import CircularProgress from 'material-ui/CircularProgress';

export default class StackLoading extends PureComponent {

  render() {
    return (
      <div style={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        flexGrow: 1,
        flexDirection: 'column'
      }}>
        <CircularProgress/>
        <span style={{marginTop: 16}}>Finding tweets...</span>
      </div>
    );
  }

}
