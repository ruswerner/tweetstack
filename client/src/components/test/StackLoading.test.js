import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import StackLoading from '../StackLoading';
import renderer from 'react-test-renderer';

jest.mock('material-ui/CircularProgress');

describe('StackLoading', function() {

  it('renders', () => {
    const tree = renderer.create(<MuiThemeProvider><StackLoading/></MuiThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
