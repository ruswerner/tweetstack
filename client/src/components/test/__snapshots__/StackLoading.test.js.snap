// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackLoading renders 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
>
  <span
    style={
      Object {
        "marginTop": 16,
      }
    }
  >
    Finding tweets...
  </span>
</div>
`;
